programs/drift/src/controller/liquidation.rs

```diff
@@ -1596,52 +1603,135 @@ pub fn resolve_perp_bankruptcy(
+                update_quote_asset_amount(
+                    perp_position,
+                    &mut market,
+                    -perp_position.quote_asset_amount,
+                )?;
+
+                market.amm.cumulative_social_loss = market
+                    .amm
+                    .cumulative_social_loss
+                    .safe_add(loss_to_socialize)?;
+
+                market.amm.cumulative_funding_rate_long =
+                    market
+                        .amm
+                        .cumulative_funding_rate_long
+                        .safe_add(cumulative_funding_rate_delta)?;
+
+                market.amm.cumulative_funding_rate_short = market
+                    .amm
+                    .cumulative_funding_rate_short
+                    .safe_sub(cumulative_funding_rate_delta)?;
+            } else {
+                delever_user_payment = if let Some(delever_user) = delever_user {
+                    let mut market = perp_market_map.get_ref_mut(&market_index)?;
+                    let free_collateral = calculate_free_collateral(
+                        delever_user,
+                        perp_market_map,
+                        spot_market_map,
+                        oracle_map,
+                        MarginRequirementType::Maintenance,
+                    )?;
+
+                    if free_collateral > 0 {
+                        let deleverage_user_position =
+                            delever_user.get_perp_position_mut(market_index).unwrap();
+                        let oracle_price_data = oracle_map.get_price_data(&market.amm.oracle)?;
+
+                        let deleverage_user_stats = DeleverageUserStats {
+                            base_asset_amount: deleverage_user_position.base_asset_amount,
+                            quote_asset_amount: deleverage_user_position.quote_asset_amount,
+                            quote_break_even_amount: deleverage_user_position
+                                .quote_break_even_amount,
+                            free_collateral,
+                        };
+
+                        let deleverage_user_payment = calculate_perp_market_deleverage_payment(
+                            loss_to_socialize,
+                            deleverage_user_stats,
+                            &market,
+                            oracle_price_data.price,
+                        )?;
+                        let perp_position =
+                            bankrupt_user.force_get_perp_position_mut(market_index)?;
+
+                        validate!(
+                            deleverage_user_payment >= 0 && perp_position.quote_asset_amount < 0,
+                            ErrorCode::DefaultError,
+                            "Deleverage User Payment to resolve bankruptcy mismatched"
+                        )?;
+
+                        update_quote_asset_amount(
```

Comment:
where do you update the deleverage user quote?

Reply:
added